dask/optimize.py

```diff
@@ -427,6 +428,179 @@ def fuse_getitem(dsk, func, place):
                            lambda a, b: tuple(b[:place]) + (a[2], ) + tuple(b[place + 1:]))
 
 
+def fuse_reductions(dsk, keys=None, ave_width=2, max_depth_new_edges=None,
+                    max_height=None, max_width=None):
```

Comment:
Done.  Note that `None` *never* means no bound (well, except for `max_height`, which has a lenient default).